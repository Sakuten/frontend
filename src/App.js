import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { inject, observer } from 'mobx-react';
import {LoginView} from './LoginView'
import {ApplicationView} from './ApplicationView'

@inject('credential')
@observer
class App extends Component {
  render() {
    return (
      <div>
        {this.props.credential.isLoggedIn ? <ApplicationView /> : <LoginView />}
      </div>
    );
  }
}

export default App;
