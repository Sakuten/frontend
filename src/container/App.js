import React from 'react'
import { inject, observer } from 'mobx-react'
import { Route, Redirect } from 'react-router'
import LoginView from './LoginView'
import ApplicationView from './ApplicationView'
import ErrorList from '../component/ErrorList'
import Home from '../component/Home'
import styled from 'styled-components'
import bg from '../sakuten.jpg'

const Container = styled.div`
  background-image: url(${bg});
  background-size: cover;

  width: 100vw;
  height: 100vh;
`

const Heading = styled.header`
  background-color: ${props => props.theme.header_color_upper};
  position: fixed;
  width: 100%;
  display: block;
  color: #000;
  z-index: 1;
  padding: 10px;
  font-size: 2rem;
  font-family: 'Roboto Condensed';
`

const App = ({store, event}) => {
  const { location } = store.router
  return (
    <div>
      <Heading>KOISHIKAWA</Heading>
      <Container location={location}>
        <Route exact path='/' component={Home} />
        <Route path='/lottery/login' render={() => store.credential.isLoggedIn ? <Redirect to='/lottery' /> : <LoginView credential={store.credential} />} />
        <Route exact path='/lottery' render={() => store.credential.isLoggedIn ? <ApplicationView user={store.credential.status} application={store.application} /> : <Redirect to='/lottery/login' />} />
        <ErrorList list={store.error.errorList} />
      </Container>
    </div>
  )
}

export default inject('event')(observer(App))
