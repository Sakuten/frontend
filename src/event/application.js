import {applyLottery, cancelLottery} from '../api/operation'

export class ApplicationObject {
  constructor (store) {
    this.store = store
    this.onUpdate()
  }

  onUpdate = async () => {
    await Promise.all([
      this.store.application.fetchClassroomList(),
      this.store.application.fetchLotteryList()
    ])
  }

  onChangeClassroom = (classroom) => {
    this.store.application.setClassroom(classroom)
  }

  onChangeLottery = (lottery) => {
    this.store.application.setLottery(lottery)
  }

  onAddGroupMember = (secretId) => {
    this.store.application.addGroupMember(secretId)
  }

  onRemoveGroupMember = (idx) => {
    this.store.application.removeGroupMemberWithIdx(idx)
  }

  onApply = async () => {
    await applyLottery(this.store.application.lottery, this.store.application.groupMemberList, this.store.credential.token)
    await this.store.fetchStatus()
  }

  onCancel = async (id) => {
    await cancelLottery(id, this.store.credential.token)
    await this.store.fetchStatus()
  }
}
