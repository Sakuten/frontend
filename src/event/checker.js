import {getPublicId, checkSecretIdStatus} from '../api/operation'
import {extractId} from '../util/extractId'

export class CheckerObject {
  constructor (store) {
    this.store = store
    this.lastScan = ''
  }

  onChangeClassroom = (classroom) => {
    this.store.checker.setClassroom(classroom)
  }

  onQRScan = async (scanUri) => {
    if (scanUri) {
      if (this.lastScan === scanUri) {
        return
      }
      this.lastScan = scanUri
      const secretId = extractId(scanUri)
      if (!secretId) {
        this.store.error.addError(101, 'Invalid QR Code')
        return
      }

      this.store.checker.setLastStatus('取得中')
      this.store.checker.setPublicId('')

      const status = await this.store.error.ignoring([19, 6], (callback) => {
        return checkSecretIdStatus(this.store.checker.classroom, secretId, this.store.credential.token)
          .then(resp => {
            return resp.data['status']
          })
          .catch(e => {
            const codeStatuses = {
              19: '応募していません',
              6: '時間外です'
            }
            return codeStatuses[e.response.data.code] || JSON.stringify(e.response.data)
          })
          .then(callback)
      })

      const resp = await getPublicId(secretId, this.store.credential.token)

      this.store.checker.setLastStatus(status)
      this.store.checker.setPublicId(resp.data['public_id'])
    }
  }
}
