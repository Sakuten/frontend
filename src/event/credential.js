import {fetchApi} from '../api'

export class CredentialObject {
  constructor(store) {
    this.store = store;
  }

  login = () => {
    fetchApi('auth/', {
      method: 'post',
      headers: {
        'Content-Type': 'application/json'
      },
      data: {
        password: this.store.credential.password,
        username: this.store.credential.username
      }
    }).then(async (response) => {
      const json = response.data
      if ('token' in json) {
        this.store.credential.setToken(json.token)
        this.store.credential.clearPassword()
        this.fetchStatus()
      } else { throw Error('Invalid response returned') }
    })
  }

  fetchStatus = () => {
    fetchApi(`api/status`, {
      method: 'get',
      headers: {
        'Authorization': 'Bearer ' + this.store.credential.token
      }
    })
      .then(response => {
        this.store.credential.setStatus(response.data.status)
      })
      .catch(error => {
        if (error.response && error.response.data.message === 'Unauthorized') {
          this.logout()
        } else {
          throw error
        }
      })
  }

  logout = () => {
    this.store.credential.setToken('')
  }

  onChangeUsername = (username) => {
    username = username.trim()
    this.store.credential.setUsername(username)
  }

  onChangePassword = (password) => {
    password = password.trim()
    this.store.credential.setPassword(password)
  }
}
