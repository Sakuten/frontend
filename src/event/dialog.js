export class DialogObject {
  constructor (store) {
    this.store = store
  }

  onClose = () => {
    this.store.dialog.close()
  }

  onOpen = (title, content, buttonText = 'OK') => {
    this.store.dialog.setTitle(title)
    this.store.dialog.setContent(content)
    this.store.dialog.setButtonText(buttonText)
    this.store.dialog.open()
  }
}
