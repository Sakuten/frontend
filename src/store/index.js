import {CredentialObject} from './credential'
import {ApplicationObject} from './application'
import { RouterStore } from 'mobx-react-router'
import {ErrorObject} from './error'

export class Store {
  constructor () {
    this.credential = new CredentialObject()
    this.application = new ApplicationObject()
    this.router = new RouterStore()
    this.error = new ErrorObject()
  }

  // Called in Event's constructor
  fetchStatus = async () => {
    await Promise.all([
      this.credential.fetchStatus(),
      this.application.fetchApplicationList(this.credential.token)
    ])
  }
}
