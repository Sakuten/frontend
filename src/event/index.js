import {CredentialObject} from './credential'
import {ApplicationObject} from './application'
import {CheckerObject} from './checker'
import {ErrorObject} from './error'
import {DialogObject} from './dialog'

import {applyLottery, cancelLottery} from '../api/operation'

export class Event {
  constructor (store) {
    this.store = store
    this.error = new ErrorObject(store)
    this.credential = new CredentialObject(store)
    this.application = new ApplicationObject(store)
    this.dialog = new DialogObject(store)
    this.checker = new CheckerObject(store)

    // This can cause network error, to be caught in ErrorObject
    // So fetchStatus() have to be placed in here
    if (this.store.credential.isLoggedIn) { this.store.fetchStatus() }
  }

  onApplyLottery = async () => {
    this.dialog.onOpen('応募しています', 'しばらくお待ちください', 'お待ちください', false)
    try {
      await applyLottery(this.store.application.lottery, this.store.application.groupMemberList.map(pair => pair[0]), this.store.credential.token)
    } catch (err) {
      this.dialog.onClose()
      throw err
    }
    this.dialog.onOpen('応募しました', '発表をお待ちください', 'OK')
    await this.store.fetchStatus()
    if (this.store.credential.isUsedByStaff) {
      this.store.credential.logout()
    }
  }

  onCancelApplication = async (id) => {
    this.dialog.onOpen('キャンセルしています', 'しばらくお待ちください', 'お待ちください', false)
    try {
      await cancelLottery(id, this.store.credential.token)
    } catch (err) {
      this.dialog.onClose()
      throw err
    }
    this.dialog.onOpen('キャンセルしました', '応募は取り消されました', 'OK')
    await this.store.fetchStatus()
    if (this.store.credential.isUsedByStaff) {
      this.store.credential.logout()
    }
  }
}
