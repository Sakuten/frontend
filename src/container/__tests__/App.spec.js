import React from 'react'
import { shallow, mount } from 'enzyme'
import deepAssign from 'deep-assign'
import { MemoryRouter } from 'react-router'
import { Provider } from 'mobx-react'
import App from '../App'
import {Store} from '../../store'
import {Event} from '../../event'

const setup = (propOverrides, storeOverrides, path = '/', isShallow = true) => {
  const store = deepAssign(new Store(), storeOverrides)
  const event = new Event(store)
  const props = Object.assign({
    store
  }, propOverrides)

  const dom = (
    <Provider event={event} >
      <MemoryRouter initialEntries={[path]}>
        <App store={store} />
      </MemoryRouter>
    </Provider>
  )
  const wrapper = (isShallow ? shallow : mount)(dom)

  return {
    props,
    wrapper,
    store,
    applicationView: wrapper.find('[data-test="applicationview"]'),
    loginView: wrapper.find('[data-test="loginview"]')
  }
}

describe('containers', () => {
  describe('App', () => {
    it('render', () => {
      const { wrapper } = setup()
      expect(wrapper).toMatchSnapshot()
    })

    it('renders ApplicationView when it is logged in', () => {
      const { loginView, applicationView } = setup({}, {credential: {token: 'token'}}, '/lottery', false)
      expect(applicationView.length).toBe(1)
      expect(loginView.length).toBe(0)
    })

    it('renders LoginView when it isn\'t logged in', () => {
      const { loginView, applicationView } = setup({}, {credential: {token: ''}}, '/lottery/login', false)
      expect(applicationView.length).toBe(0)
      expect(loginView.length).toBe(1)
    })

    it('redirects from /lottery to /lottery/login when it isn\'t logged in', () => {
      const { wrapper } = setup({}, {credential: {token: ''}}, '/lottery', false)
      expect(wrapper.find('Redirect').prop('to')).toBe('/lottery/login')
    })

    it('redirects from /lottery/login to /lottery when it is logged in', () => {
      const { wrapper } = setup({}, {credential: {token: 'token'}}, '/lottery/login', false)
      expect(wrapper.find('Redirect').prop('to')).toBe('/lottery')
    })
  })
})
