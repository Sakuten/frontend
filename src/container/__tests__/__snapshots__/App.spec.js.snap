// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers App render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    event={
      Event {
        "application": ApplicationObject {
          "onApply": [Function],
          "onCancel": [Function],
          "onChangeClassroom": [Function],
          "onChangeLottery": [Function],
          "onUpdate": [Function],
          "store": Store {
            "application": ApplicationObject {},
            "credential": CredentialObject {
              "recaptchaResponse": "",
              "status": Object {},
              "token": "",
              "username": "",
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "CredentialObject@3.keys",
                  "observers": Set {},
                },
                "name": "CredentialObject@3",
                "target": [Circular],
                "values": Map {
                  "username" => "",
                  "recaptchaResponse" => "",
                  "token" => "",
                  "status" => Map {},
                  "isLoggedIn" => false,
                },
              },
            },
            "error": ErrorObject {},
            "fetchStatus": [Function],
            "router": RouterStore {
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "history": null,
              "push": [Function],
              "replace": [Function],
            },
          },
        },
        "credential": CredentialObject {
          "onChangeRecaptchaResponse": [Function],
          "onChangeUsername": [Function],
          "onLogin": [Function],
          "onLogout": [Function],
          "store": Store {
            "application": ApplicationObject {},
            "credential": CredentialObject {
              "recaptchaResponse": "",
              "status": Object {},
              "token": "",
              "username": "",
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "CredentialObject@3.keys",
                  "observers": Set {},
                },
                "name": "CredentialObject@3",
                "target": [Circular],
                "values": Map {
                  "username" => "",
                  "recaptchaResponse" => "",
                  "token" => "",
                  "status" => Map {},
                  "isLoggedIn" => false,
                },
              },
            },
            "error": ErrorObject {},
            "fetchStatus": [Function],
            "router": RouterStore {
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "history": null,
              "push": [Function],
              "replace": [Function],
            },
          },
        },
        "error": ErrorObject {
          "onError": [Function],
          "store": Store {
            "application": ApplicationObject {},
            "credential": CredentialObject {
              "recaptchaResponse": "",
              "status": Object {},
              "token": "",
              "username": "",
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "CredentialObject@3.keys",
                  "observers": Set {},
                },
                "name": "CredentialObject@3",
                "target": [Circular],
                "values": Map {
                  "username" => "",
                  "recaptchaResponse" => "",
                  "token" => "",
                  "status" => Map {},
                  "isLoggedIn" => false,
                },
              },
            },
            "error": ErrorObject {},
            "fetchStatus": [Function],
            "router": RouterStore {
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "history": null,
              "push": [Function],
              "replace": [Function],
            },
          },
        },
        "store": Store {
          "application": ApplicationObject {},
          "credential": CredentialObject {
            "recaptchaResponse": "",
            "status": Object {},
            "token": "",
            "username": "",
            Symbol(mobx did run lazy initializers): true,
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "CredentialObject@3.keys",
                "observers": Set {},
              },
              "name": "CredentialObject@3",
              "target": [Circular],
              "values": Map {
                "username" => "",
                "recaptchaResponse" => "",
                "token" => "",
                "status" => Map {},
                "isLoggedIn" => false,
              },
            },
          },
          "error": ErrorObject {},
          "fetchStatus": [Function],
          "router": RouterStore {
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "history": null,
            "push": [Function],
            "replace": [Function],
          },
        },
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          "/",
        ]
      }
    >
      <inject-App-with-event
        store={
          Store {
            "application": ApplicationObject {},
            "credential": CredentialObject {
              "recaptchaResponse": "",
              "status": Object {},
              "token": "",
              "username": "",
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "CredentialObject@3.keys",
                  "observers": Set {},
                },
                "name": "CredentialObject@3",
                "target": [Circular],
                "values": Map {
                  "username" => "",
                  "recaptchaResponse" => "",
                  "token" => "",
                  "status" => Map {},
                  "isLoggedIn" => false,
                },
              },
            },
            "error": ErrorObject {},
            "fetchStatus": [Function],
            "router": RouterStore {
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "history": null,
              "push": [Function],
              "replace": [Function],
            },
          }
        }
      />
    </MemoryRouter>
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <inject-App-with-event
        store={
          Store {
            "application": ApplicationObject {},
            "credential": CredentialObject {
              "recaptchaResponse": "",
              "status": Object {},
              "token": "",
              "username": "",
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "CredentialObject@3.keys",
                  "observers": Set {},
                },
                "name": "CredentialObject@3",
                "target": [Circular],
                "values": Map {
                  "username" => "",
                  "recaptchaResponse" => "",
                  "token" => "",
                  "status" => Map {},
                  "isLoggedIn" => false,
                },
              },
            },
            "error": ErrorObject {},
            "fetchStatus": [Function],
            "router": RouterStore {
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "history": null,
              "push": [Function],
              "replace": [Function],
            },
          }
        }
      />,
      "initialEntries": Array [
        "/",
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "store": Store {
          "application": ApplicationObject {},
          "credential": CredentialObject {
            "recaptchaResponse": "",
            "status": Object {},
            "token": "",
            "username": "",
            Symbol(mobx did run lazy initializers): true,
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "CredentialObject@3.keys",
                "observers": Set {},
              },
              "name": "CredentialObject@3",
              "target": [Circular],
              "values": Map {
                "username" => "",
                "recaptchaResponse" => "",
                "token" => "",
                "status" => Map {},
                "isLoggedIn" => false,
              },
            },
          },
          "error": ErrorObject {},
          "fetchStatus": [Function],
          "router": RouterStore {
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "history": null,
            "push": [Function],
            "replace": [Function],
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <inject-App-with-event
          store={
            Store {
              "application": ApplicationObject {},
              "credential": CredentialObject {
                "recaptchaResponse": "",
                "status": Object {},
                "token": "",
                "username": "",
                Symbol(mobx did run lazy initializers): true,
                Symbol(mobx administration): ObservableObjectAdministration$$1 {
                  "defaultEnhancer": [Function],
                  "keysAtom": Atom$$1 {
                    "diffValue": 0,
                    "isBeingObserved": false,
                    "isPendingUnobservation": false,
                    "lastAccessedBy": 0,
                    "lowestObserverState": 2,
                    "name": "CredentialObject@3.keys",
                    "observers": Set {},
                  },
                  "name": "CredentialObject@3",
                  "target": [Circular],
                  "values": Map {
                    "username" => "",
                    "recaptchaResponse" => "",
                    "token" => "",
                    "status" => Map {},
                    "isLoggedIn" => false,
                  },
                },
              },
              "error": ErrorObject {},
              "fetchStatus": [Function],
              "router": RouterStore {
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "history": null,
                "push": [Function],
                "replace": [Function],
              },
            }
          }
        />,
        "initialEntries": Array [
          "/",
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "store": Store {
            "application": ApplicationObject {},
            "credential": CredentialObject {
              "recaptchaResponse": "",
              "status": Object {},
              "token": "",
              "username": "",
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "CredentialObject@3.keys",
                  "observers": Set {},
                },
                "name": "CredentialObject@3",
                "target": [Circular],
                "values": Map {
                  "username" => "",
                  "recaptchaResponse" => "",
                  "token" => "",
                  "status" => Map {},
                  "isLoggedIn" => false,
                },
              },
            },
            "error": ErrorObject {},
            "fetchStatus": [Function],
            "router": RouterStore {
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "history": null,
              "push": [Function],
              "replace": [Function],
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
